tcms/core/middleware.py

```diff
@@ -22,9 +23,12 @@ def process_request(self, request):
             messages.add_message(
                 request,
                 messages.ERROR,
-                _('Base URL is not configured! '
-                  'See <a href="%(doc_url)s">documentation</a> and '
-                  '<a href="%(admin_url)s">change it</a>') % {
-                      'doc_url': doc_url,
-                      'admin_url': reverse('admin:sites_site_change', args=[site.pk])}
+                mark_safe(
+                    _('Base URL is not configured! '
+                      'See <a href="%(doc_url)s">documentation</a> and '
+                      '<a href="%(admin_url)s">change it</a>') % {
+                          'doc_url': doc_url,
+                          'admin_url': reverse('admin:sites_site_change', args=[site.pk])
+                      }
+                )
```

Comment:
Missing blank line at the end of the file.

Reply:
an extra blank line is not required. A newline is!